.github/workflows/tuist.yml

```diff
@@ -0,0 +1,57 @@
+name: Lint
+
+on:
+  push:
+    branches:
+      - main
+  pull_request:
+    paths:
+      - "Sources/**"
+      - "Tests/**"
+      - "*.swift"
+      - "Package.resolved"
+      - ".github/workflows/tuist.yml"
+      - "mise/tasks/lint"
+      - "mise/tasks/lint-fix"
+      - "mise.toml"
+      - "fixtures/**"
+      - "Tuist/**"
+
+permissions:
+  contents: read
+
+env:
+  TUIST_CONFIG_TOKEN: ${{ secrets.TUIST_CONFIG_TOKEN }}
+  MISE_SOPS_AGE_KEY: ${{ secrets.MISE_SOPS_AGE_KEY }}
+  GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}
+
+concurrency:
+  group: tuist-${{ github.head_ref }}
+  cancel-in-progress: true
+
+jobs:
+  lint:
+    runs-on: macos-15
+    steps:
+      - uses: actions/checkout@v4
+      - name: Select Xcode
+        run: sudo xcode-select -switch /Applications/Xcode_$(cat .xcode-version).app
+      - name: Restore cache
```

Comment:
It would be awesome if the caching of these was done automatically by Tuist. Should we remove the .git directories to speed up this?